singlestoredb/management/job.py

```diff
@@ -0,0 +1,792 @@
+            Mode.ONCE,
+            False,
+            start_at=datetime.datetime.now(),
+            runtime_name=runtime_name,
+            parameters=parameters,
+        )
+
+    def wait(self, jobs: List[Union[str, Job]], timeout: Optional[int] = None) -> bool:
+        if timeout is not None:
+            if timeout <= 0:
+                return False
+            finish_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+
+        for job in jobs:
+            if timeout is not None:
+                job_timeout = int((finish_time - datetime.datetime.now()).total_seconds())
+            else:
+                job_timeout = None
+
+            res = self._wait_for_job(job, job_timeout)
+            if not res:
+                return False
+
+        return True
+
+    def _wait_for_job(self, job: Union[str, Job], timeout: Optional[int] = None) -> bool:
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+
+        if timeout is not None:
+            if timeout <= 0:
+                return False
+            finish_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+
+        if isinstance(job, str):
+            job_id = job
+        else:
+            job_id = job.job_id
+
+        while True:
+            if timeout is not None and datetime.datetime.now() >= finish_time:
+                return False
+
+            res = self._manager._get(f'jobs/{job_id}').json()
+            job = Job.from_dict(res, self)
+            if job.schedule.mode == Mode.ONCE and job.completed_executions_count > 0:
+                return True
+            if job.schedule.mode == Mode.RECURRING:
+                raise ValueError(f'Cannot wait for recurring job {job_id}')
+            time.sleep(1)
+
+    def get(self, job_id: str) -> Job:
+        """Get a job by its ID."""
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+
+        res = self._manager._get(f'jobs/{job_id}').json()
+        return Job.from_dict(res, self)
+
+    def get_executions(
+            self,
+            job_id: str,
+            start_execution_number: int,
+            end_execution_number: int,
+    ) -> ExecutionsData:
+        """Get executions for a job by its ID."""
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+        path = (
+            f'jobs/{job_id}/executions'
+            f'?start={start_execution_number}'
+            f'&end={end_execution_number}'
+        )
+        res = self._manager._get(path).json()
+        return ExecutionsData.from_dict(res)
+
+    def delete(self, job_id: str) -> bool:
+        """Delete a job by its ID."""
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+
+        return self._manager._delete(f'jobs/{job_id}').json()
+
+    def modes(self) -> Type[Mode]:
```

Comment:
@kesmit13 what do you think about the `Enums` I added in this file, e.g. for Mode,? I didn't see any other `Enums` in the code and I wondered if that is the best way to do it, or if there is a better one.

What I found is that if I have an `Enum` as the input type, it kind of forces the user to import the `Enum` to get the possible values. Or we need to implement something like the `modes` function.

Reply:
I think I just used strings for enum-type things because the Python API sometimes lags the Management API and if there are values added to the Management API that aren't reflected in the Python API then you'll eventually get errors. I agree that it is nicer to have some validation in the Python code to prevent bad values from getting in, but I err'd on the side of keeping APIs working when there may be changes.

Reply:
Maybe I should also change to strings to prevent having to do MRs to change the python SDK when there are changes to these values.

Reply:
I decided to keep `Mode` as an `Enum` since the values are not expected to changes, but `runtime_name` will be kept as a `string` to support addition of new Nova pools without additional MRs on the python SDK.